server/api/post/post.controller.js

```diff
@@ -45,7 +45,7 @@ exports.create = function(req, res) {
     if(!project) { return res.status(404).send("Project not found"); }
 
     // Only someone who is part of the project can write a blog post
-    User.findById(req.user._id, function(err, user) {
+    User.findById(req.user._id,  '-salt -hashedPassword', function(err, user) {
```

Comment:
the -salt and -hashedPassword should be hidden by the model and you shouldn't need to explicitly removed